dbms/src/Storages/Tantivy/TantivyInputStream.h

```diff
@@ -117,23 +118,15 @@ class TantivyInputStream : public IProfilingBlockInputStream
         Block res(return_columns);
         if (is_count)
         {
-            if (search_result.count == 0)
-            {
-                return res;
-            }
-            // Construct const columns with search_result.count items, only used by count(*)
-            for (auto & name_and_type : return_columns)
-            {
-                auto type_default_value = name_and_type.type->getDefault();
-                auto col = name_and_type.type->createColumnConst(
-                    static_cast<size_t>(search_result.count),
-                    type_default_value);
-                res.getByName(name_and_type.name).column = std::move(col);
-            }
-            return res;
+            RUNTIME_ASSERT(return_columns.size() == 1, log, "count search should return one column");
+            Block result_block;
```

Comment:
You can directly use `res` at line 118, and there’s no need to define and initialize `result_block`